Status: unsubmitted

This patch was for Debian bug #239535.  It needs to be tested, and
submitted.

---
 gdb/dwarf2read.c |    2 +-
 gdb/valops.c     |    4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

Index: gdb-6.8.50.20081120.python/gdb/valops.c
===================================================================
--- gdb-6.8.50.20081120.python.orig/gdb/valops.c	2008-11-25 16:16:41.000000000 -0500
+++ gdb-6.8.50.20081120.python/gdb/valops.c	2008-11-26 17:14:38.000000000 -0500
@@ -2472,8 +2472,10 @@ check_field (struct type *type, const ch
 	return 1;
     }
 
+  /* Check each baseclass.  Call check_typedef, which will follow typedefs
+     and do opaque/stub type resolution.  */
   for (i = TYPE_N_BASECLASSES (type) - 1; i >= 0; i--)
-    if (check_field (TYPE_BASECLASS (type, i), name))
+    if (check_field (check_typedef (TYPE_BASECLASS (type, i)), name))
       return 1;
 
   return 0;
Index: gdb-6.8.50.20081120.python/gdb/dwarf2read.c
===================================================================
--- gdb-6.8.50.20081120.python.orig/gdb/dwarf2read.c	2008-11-25 16:16:41.000000000 -0500
+++ gdb-6.8.50.20081120.python/gdb/dwarf2read.c	2008-11-26 17:14:06.000000000 -0500
@@ -2273,8 +2273,8 @@ guess_structure_name (struct partial_die
 				    strlen (actual_class_name),
 				    &cu->comp_unit_obstack);
 		  xfree (actual_class_name);
+		  break;
 		}
-	      break;
 	    }
 
 	  child_pdi = child_pdi->die_sibling;
